scipy/stats/stats.py

```diff
@@ -6143,12 +6147,26 @@ def power_divergence(f_obs, f_exp=None, ddof=0, axis=0, lambda_=None):
         with np.errstate(invalid='ignore'):
             f_exp = f_obs.mean(axis=axis, keepdims=True)
 
+    f_obs_float = f_obs.astype(np.float64)
+    rtol = 1e-8  # to pass existing tests
+    with np.errstate(invalid='ignore'):
+        relative_diff = np.abs((f_obs_float - f_exp).sum(axis=axis)
+                               / np.minimum(f_obs_float, f_exp).sum(axis=axis))
```

Comment:
I think this should be
```suggestion
                               / np.minimum(f_obs_float.sum(axis=axis), f_exp.sum(axis=axis))
```
This makes the rel. diff. equal to  `abs(N - M)/min(N, M)` where `N` is the sum of the observed frequencies and `M` is the sum of the expected frequencies.  Taking the minimum before summing does something different.

Reply:
It was actually intentional. The sum of the minimums is always less than or equal to the minimum of the sums, so the relative error calculated is conservative (high). Since the tolerance is arbitrary, I figured it was OK.

If you want it the other way, the arrays have to be broadcasted and the masks need to be shared before the sum is taken.

Reply:
Ah, I see.  It would be nice if we can stick to a standard definition of "relative error", and `abs(N-M)/min(N,M)` is fairly standard (e.g. that's what `math.isclose()` does (in effect) for its relative tolerance).

We can avoid some of the issues by only doing this test when `f_exp` is not None, and with a few other tweaks, all the test pass while using the more familiar version of "relative error".  I replaced lines 6139-6163 with this:
```
    f_obs = np.asanyarray(f_obs)
    f_obs_float = f_obs.astype(np.float64)

    if f_exp is not None:
        f_exp = np.broadcast_to(np.asanyarray(f_exp), f_obs.shape, subok=True)

        rtol = 1e-8  # to pass existing tests
        with np.errstate(invalid='ignore'):
            f_obs_sum = f_obs_float.sum(axis=axis)
            f_exp_sum = f_exp.sum(axis=axis)
            relative_diff = (np.abs(f_obs_sum - f_exp_sum) /
                             np.minimum(f_obs_sum, f_exp_sum))
            diff_gt_tol = (relative_diff > rtol).any()
        if diff_gt_tol:
            msg = (f"For each axis slice, the sum of the observed "
                   f"frequencies must agree with the sum of the "
                   f"expected frequencies to a relative tolerance "
                   f"of {rtol}, but the percent differences are:\n"
                   f"{relative_diff}")
            raise ValueError(msg)
    else:
        # Ignore 'invalid' errors so the edge case of a data set with length 0
        # is handled without spurious warnings.
        with np.errstate(invalid='ignore'):
            f_exp = f_obs.mean(axis=axis, keepdims=True)
```
Maybe that could be simplified a bit (e.g. not sure we really need both `f_obs` and `f_obs_float`).

Reply:
Looks very similar to what I had originally. 
Any chance that `f_exp` is masked? `broadcast_to` strips it.

Reply:
I ran into that, but then noticed that if you add the argument `subok=True` to `broadcast_to`, it preserves the masked arrays.

Reply:
```
> f_exp = np.ma.MaskedArray([1, 2, 3, 4, 5], mask = [0, 1, 0, 1, 0])
> f_exp = np.broadcast_to(np.asanyarray(f_exp), (3,5), subok=True)
> f_exp
masked_array(
  data=[[1, 2, 3, 4, 5],
        [1, 2, 3, 4, 5],
        [1, 2, 3, 4, 5]],
  mask=False,
  fill_value=999999)
```

Reply:
We can do the obvious thing - broadcast the mask manually - I was just trying to avoid all this mess.

Reply:
Sigh. ☹️

Reply:
> We can do the obvious thing - broadcast the mask manually...

Maybe something like this:

```
def m_broadcast_to(a, shape): 
    if np.ma.isMaskedArray(a): 
        return np.ma.masked_array(np.broadcast_to(a, shape), mask=np.broadcast_to(a.mask, shape))
    return np.broadcast_to(a, shape)
```

Reply:
Done but do you agree this is something that belongs in NumPy? Maybe this isn't the first time somebody has needed to broadcast masked arrays.

Reply:
See refguide check. I guess that's why I used `broadcast` on both arrays when I first tried this.